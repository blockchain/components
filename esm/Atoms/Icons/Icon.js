function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  height: ", ";\n  font-weight: ", ";\n  font-size: ", ";\n  color: ", ";\n  cursor: ", ";\n\n  &:hover {\n    color: ", ";\n  }\n\n  &::before {\n    font-family: 'icomoon';\n    content: '", "';\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import IcomoonMap from './IcomoonMap';

var getHoverColor = function getHoverColor(selectable, theme) {
  return selectable ? theme['iconHoverColor'] : theme['iconColor'];
};

var BaseIcon = styled.span(_templateObject(), function (props) {
  return props.size;
}, function (props) {
  return props.weight;
}, function (props) {
  return props.size;
}, function (props) {
  return props.theme['iconColor'];
}, function (props) {
  return props.selectable ? 'pointer' : 'inherit';
}, function (props) {
  return getHoverColor(props.selectable, props.theme);
}, function (props) {
  return props.code;
});

var Icon = function Icon(_ref) {
  var name = _ref.name,
      selectable = _ref.selectable,
      rest = _objectWithoutProperties(_ref, ["name", "selectable"]);

  return React.createElement(BaseIcon, _extends({}, rest, {
    code: IcomoonMap[name],
    selectable: selectable
  }));
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  selectable: PropTypes.bool,
  size: PropTypes.string,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900])
};
Icon.defaultProps = {
  name: 'bitcoin',
  selectable: false,
  size: '40px',
  weight: 400
};
export default Icon;