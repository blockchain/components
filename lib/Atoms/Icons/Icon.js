"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _IcomoonMap = _interopRequireDefault(require("./IcomoonMap"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  height: ", ";\n  font-weight: ", ";\n  font-size: ", ";\n  color: ", ";\n  cursor: ", ";\n\n  &:hover {\n    color: ", ";\n  }\n\n  &::before {\n    font-family: 'icomoon';\n    content: '", "';\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var getHoverColor = function getHoverColor(selectable, theme) {
  return selectable ? theme['iconHoverColor'] : theme['iconColor'];
};

var BaseIcon = _styledComponents.default.span(_templateObject(), function (props) {
  return props.size;
}, function (props) {
  return props.weight;
}, function (props) {
  return props.size;
}, function (props) {
  return props.theme['iconColor'];
}, function (props) {
  return props.selectable ? 'pointer' : 'inherit';
}, function (props) {
  return getHoverColor(props.selectable, props.theme);
}, function (props) {
  return props.code;
});

var Icon = function Icon(_ref) {
  var name = _ref.name,
      selectable = _ref.selectable,
      rest = _objectWithoutProperties(_ref, ["name", "selectable"]);

  return _react.default.createElement(BaseIcon, _extends({}, rest, {
    code: _IcomoonMap.default[name],
    selectable: selectable
  }));
};

Icon.propTypes = {
  name: _propTypes.default.string.isRequired,
  selectable: _propTypes.default.bool,
  size: _propTypes.default.string,
  weight: _propTypes.default.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900])
};
Icon.defaultProps = {
  name: 'bitcoin',
  selectable: false,
  size: '40px',
  weight: 400
};
var _default = Icon;
exports.default = _default;