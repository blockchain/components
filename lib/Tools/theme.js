"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _palette = _interopRequireDefault(require("./palette"));

var _polished = require("polished");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable sort-keys */
var _default = {
  fontPrimary: "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
  buttonColor: _palette.default['white'],
  buttonBackgroundColor: _palette.default['orient'],
  buttonHoverColor: (0, _polished.lighten)(0.1, _palette.default['orient']),
  cartridgeColor: _palette.default['white'],
  cartridgeBackgroundColor: _palette.default['orient'],
  circleBadgeBackgroundColor: _palette.default['gray2'],
  circleBadgeHoverColor: _palette.default['orient'],
  dropdownColor: _palette.default['gray5'],
  dropdownBackgroundColor: _palette.default['transparent'],
  iconColor: _palette.default['gray5'],
  iconHoverColor: _palette.default['gray4'],
  inputColor: _palette.default['gray5'],
  inputBackgroundColor: _palette.default['white'],
  inputBorderColor: _palette.default['gray2'],
  linkColor: _palette.default['gray5'],
  linkHoverColor: _palette.default['orient'],
  panelBackgroundColor: _palette.default['white'],
  panelShadowColor: _palette.default['gray1'],
  searchBarInputBackgroundColor: _palette.default['white'],
  searchBarIconBackgroundColor: _palette.default['gold'],
  searchBarUnderlinedBackgroundColor: _palette.default['transparent'],
  searchBarUnderlinedBorderColor: _palette.default['gray2'],
  sectionBackgroundColor: _palette.default['orient'],
  tableShadowColor: _palette.default['gray1'],
  tableHeaderColor: _palette.default['gray3'],
  tableCellEvenBackgroundColor: _palette.default['gray0'],
  tableCellOddBackgroundColor: _palette.default['white'],
  tabMenuBorderColor: _palette.default['gray1'],
  tabMenuBarColor: _palette.default['gold'],
  textColor: _palette.default['gray5']
  /* eslint-enable */

};
exports.default = _default;