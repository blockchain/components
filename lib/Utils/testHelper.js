"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}