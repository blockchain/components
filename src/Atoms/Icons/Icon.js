import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Text from './../Typography/Text'
import IcomoonMap from './IcomoonMap'

const getHoverColor = (selectable, theme) =>
  selectable ? theme['iconHoverColor'] : theme['iconColor']

const Wrapper = styled(Text)`
  color: ${(props) => props.theme['iconColor']};
  cursor: ${(props) => (props.selectable ? 'pointer' : 'inherit')};

  &:hover {
    color: ${(props) => getHoverColor(props.selectable, props.theme)};
  }

  &::before {
    font-family: 'icomoon';
    content: '${(props) => props.code}';
  }
`

const Icon = ({ name, ...rest }) => (
  <Wrapper {...rest} code={IcomoonMap[name]} />
)

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  selectable: PropTypes.bool,
}

Icon.defaultProps = {
  name: 'bitcoin',
  selectable: false,
  size: '40px',
  weight: 400,
}

export default Icon
