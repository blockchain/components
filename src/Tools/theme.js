import palette from './palette'
import { lighten } from 'polished'
/* eslint-disable sort-keys */
export default {
  fontPrimary: "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
  buttonColor: palette['white'],
  buttonBackgroundColor: palette['orient'],
  buttonHoverColor: lighten(0.1, palette['orient']),
  cartridgeColor: palette['white'],
  cartridgeBackgroundColor: palette['orient'],
  circleBadgeBackgroundColor: palette['gray2'],
  circleBadgeHoverColor: palette['orient'],
  dropdownColor: palette['gray5'],
  dropdownBackgroundColor: palette['transparent'],
  iconColor: palette['gray5'],
  iconHoverColor: palette['gray4'],
  inputColor: palette['gray5'],
  inputBackgroundColor: palette['white'],
  inputBorderColor: palette['gray2'],
  linkColor: palette['gray5'],
  linkHoverColor: palette['orient'],
  panelBackgroundColor: palette['white'],
  panelShadowColor: palette['gray1'],
  searchBarInputBackgroundColor: palette['white'],
  searchBarIconBackgroundColor: palette['gold'],
  searchBarUnderlinedBackgroundColor: palette['transparent'],
  searchBarUnderlinedBorderColor: palette['gray2'],
  sectionBackgroundColor: palette['orient'],
  tableShadowColor: palette['gray1'],
  tableHeaderColor: palette['gray3'],
  tableCellEvenBackgroundColor: palette['gray0'],
  tableCellOddBackgroundColor: palette['white'],
  tabMenuBorderColor: palette['gray1'],
  tabMenuBarColor: palette['gold'],
  textColor: palette['gray5'],
}
/* eslint-enable */
