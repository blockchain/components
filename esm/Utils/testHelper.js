import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}