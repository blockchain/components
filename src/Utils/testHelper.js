import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
