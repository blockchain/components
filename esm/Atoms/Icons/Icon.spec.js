import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import theme from '../../Tools/theme'
import Icon from './Icon'

describe('Icon theme={theme}', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Icon />)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with name', () => {
    const tree = mountWithTheme(<Icon name="ethereum" />)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mountWithTheme(<Icon size="72px" />)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with uppercase', () => {
    const tree = mountWithTheme(<Icon weight={400} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with selectable', () => {
    const tree = mountWithTheme(<Icon selectable />)

    expect(tree).toMatchSnapshot()
  })
})
