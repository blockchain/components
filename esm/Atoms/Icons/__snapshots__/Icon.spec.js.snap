// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon theme={theme} should render correctly 1`] = `
.c0 {
  height: 40px;
  font-weight: 400;
  font-size: 40px;
  color: #545456;
  cursor: inherit;
}

.c0:hover {
  color: #545456;
}

.c0::before {
  font-family: 'icomoon';
  content: '\\e957';
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Icon
    name="bitcoin"
    selectable={false}
    size="40px"
    weight={400}
  >
    <styled.span
      code="\\\\e957"
      selectable={false}
      size="40px"
      weight={400}
    >
      <span
        className="c0"
        size="40px"
      />
    </styled.span>
  </Icon>
</ThemeProvider>
`;

exports[`Icon theme={theme} should render correctly with name 1`] = `
.c0 {
  height: 40px;
  font-weight: 400;
  font-size: 40px;
  color: #545456;
  cursor: inherit;
}

.c0:hover {
  color: #545456;
}

.c0::before {
  font-family: 'icomoon';
  content: '\\e963';
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Icon
    name="ethereum"
    selectable={false}
    size="40px"
    weight={400}
  >
    <styled.span
      code="\\\\e963"
      selectable={false}
      size="40px"
      weight={400}
    >
      <span
        className="c0"
        size="40px"
      />
    </styled.span>
  </Icon>
</ThemeProvider>
`;

exports[`Icon theme={theme} should render correctly with selectable 1`] = `
.c0 {
  height: 40px;
  font-weight: 400;
  font-size: 40px;
  color: #545456;
  cursor: pointer;
}

.c0:hover {
  color: #757679;
}

.c0::before {
  font-family: 'icomoon';
  content: '\\e957';
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Icon
    name="bitcoin"
    selectable={true}
    size="40px"
    weight={400}
  >
    <styled.span
      code="\\\\e957"
      selectable={true}
      size="40px"
      weight={400}
    >
      <span
        className="c0"
        size="40px"
      />
    </styled.span>
  </Icon>
</ThemeProvider>
`;

exports[`Icon theme={theme} should render correctly with size 1`] = `
.c0 {
  height: 72px;
  font-weight: 400;
  font-size: 72px;
  color: #545456;
  cursor: inherit;
}

.c0:hover {
  color: #545456;
}

.c0::before {
  font-family: 'icomoon';
  content: '\\e957';
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Icon
    name="bitcoin"
    selectable={false}
    size="72px"
    weight={400}
  >
    <styled.span
      code="\\\\e957"
      selectable={false}
      size="72px"
      weight={400}
    >
      <span
        className="c0"
        size="72px"
      />
    </styled.span>
  </Icon>
</ThemeProvider>
`;

exports[`Icon theme={theme} should render correctly with uppercase 1`] = `
.c0 {
  height: 40px;
  font-weight: 400;
  font-size: 40px;
  color: #545456;
  cursor: inherit;
}

.c0:hover {
  color: #545456;
}

.c0::before {
  font-family: 'icomoon';
  content: '\\e957';
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Icon
    name="bitcoin"
    selectable={false}
    size="40px"
    weight={400}
  >
    <styled.span
      code="\\\\e957"
      selectable={false}
      size="40px"
      weight={400}
    >
      <span
        className="c0"
        size="40px"
      />
    </styled.span>
  </Icon>
</ThemeProvider>
`;
